src/Workspaces/MSBuildTest/OpenProjectsTests.cs

```diff
@@ -0,0 +1,222 @@
+        {
+            var ignoredDiagnostics = templateName switch
+            {
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.CSharp, ignoredDiagnostics);
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled), typeof(WindowsOnly))]
+        [InlineData("winforms")]
+        [InlineData("winformslib")]
+        [InlineData("winformscontrollib")]
+        [InlineData("wpf")]
+        [InlineData("wpflib")]
+        [InlineData("wpfcustomcontrollib")]
+        [InlineData("wpfusercontrollib")]
+        public async Task VisualBasicTemplateProject_WindowsOnly_LoadWithNoDiagnostics(string templateName)
+        {
+            var ignoredDiagnostics = templateName switch
+            {
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.VisualBasic, ignoredDiagnostics);
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled))]
+        [InlineData("classlib")]
+        [InlineData("console")]
+        [InlineData("mstest")]
+        [InlineData("nunit")]
+        [InlineData("xunit")]
+        public async Task VisualBasicTemplateProject_LoadWithNoDiagnostics(string templateName)
+        {
+            var ignoredDiagnostics = (templateName, isWindows: ExecutionConditionUtil.IsWindows) switch
+            {
+                (_, isWindows: false) => new string[] { ERR_UndefinedType1 },
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.VisualBasic, ignoredDiagnostics);
+        }
+
+        private async Task AssertTemplateProjectLoadsCleanlyAsync(string templateName, string languageName, string[] ignoredDiagnostics)
+        {
+            try
+            {
+                if (ignoredDiagnostics.Length > 0)
+                {
+                    TestOutputHelper.WriteLine($"Ignoring compiler diagnostics: \"{string.Join("\", \"", ignoredDiagnostics)}\"");
+                }
+
+                // Clean up previous run
+                CleanupProject(templateName, languageName);
+
+                var projectFilePath = GenerateProjectFromTemplate(templateName, languageName, TestOutputHelper);
+
+                await AssertProjectLoadsCleanlyAsync(projectFilePath, ignoredDiagnostics);
+
+                // Clean up successful run
+                CleanupProject(templateName, languageName);
+            }
+            catch
+            {
+                throw;
+            }
+        }
+
+        private string GenerateProjectFromTemplate(string templateName, string languageName, ITestOutputHelper outputHelper)
+        {
+            var projectPath = GetProjectPath(templateName, languageName);
+            var projectFilePath = GetProjectFilePath(projectPath, languageName);
```

Comment:
It's not entirely clear to me why we're producing subfolders here since I'd expect the temporary directory to be empty from the start.

Reply:
In dotnet-format we create subfolder because we don't delete the folder if the project fails to load and instead upload it as an artifact so we can debug through it. We don't have to keep that behavior here.

Reply:
Updated to use the SolutionDirectory directly.